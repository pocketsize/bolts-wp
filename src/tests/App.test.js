import React from 'react';
import { shallow, mount } from 'enzyme';
import App from './App';

// smoke test - test if it render
test('it should render', () => {
	shallow(<App />);
});

// snapshot test - test the final HTML
test('it should render the expected HTML', () => {
	expect(
		mount(<App />).html()
	).toMatchSnapshot();
});

// unit test - test functionality
test('it should update the state and change the DOM after each click', () => {
	const wrapper = shallow(<App />);

	expect(
		wrapper.find('strong').text()
	).toBe('0');

	wrapper.find('div').simulate('click');

	expect(
		wrapper.find('strong').text()
	).toBe('1');
});