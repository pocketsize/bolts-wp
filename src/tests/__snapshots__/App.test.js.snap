// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render the expected HTML 1`] = `"<div>Clicked <strong>0</strong> times!</div>"`;
